rotkehlchen/chain/evm/decoding/curve_lend/decoder.py

```diff
@@ -0,0 +1,617 @@
+            msg_aggregator: 'MessagesAggregator',
+            leverage_zap: 'ChecksumEvmAddress | None',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.leverage_zap = leverage_zap
+        self.vaults: set[ChecksumEvmAddress] = set()
+        self.controllers: set[ChecksumEvmAddress] = set()
+
+    def reload_data(self) -> Mapping['ChecksumEvmAddress', tuple[Any, ...]] | None:
+        """Check that cache is up to date and refresh cache from db.
+        Returns a fresh addresses to decoders mapping."""
+        if should_update_protocol_cache(CacheType.CURVE_VAULTS) is True:
+            query_curve_vaults(database=self.evm_inquirer.database)
+        elif len(self.vaults) != 0:
+            return None  # we didn't update the globaldb cache, and we have the data already
+
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            query_body = 'FROM evm_tokens WHERE protocol=? AND chain=?'
+            bindings = (CURVE_VAULTS_PROTOCOL, self.evm_inquirer.chain_id.serialize_for_db())
+
+            cursor.execute(f'SELECT COUNT(*) {query_body}', bindings)
+            if cursor.fetchone()[0] == len(self.vaults):
+                return None  # up to date
+
+            # we are missing new vaults. Populate the cache
+            vault_data = cursor.execute(f'SELECT address {query_body}', bindings).fetchall()
+
+        for row in vault_data:
+            if (controller_address := self._get_possibly_cached_vault_address(
+                cache_type=CacheType.CURVE_VAULT_CONTROLLER,
+                vault_address=(vault_address := row[0]),
+                contract_method='controller',
+            )) is None:
+                log.error(f'Failed to load controller address for Curve vault {vault_address}')
+                continue
+
+            self.vaults.add(vault_address)
+            self.controllers.add(controller_address)
+
+        return self.addresses_to_decoders()
+
+    def _get_vault_event_tokens_and_amounts(
+            self,
+            vault_address: 'ChecksumEvmAddress',
+            context: DecoderContext,
+    ) -> tuple['EvmToken', 'EvmToken', 'FVal', 'FVal'] | None:
+        """Get the vault token, underlying token, and the corresponding amounts.
+        Returns the tokens and amounts in a tuple or None on error."""
+        try:
+            vault_token = self.base.get_or_create_evm_token(address=vault_address)
+            if vault_token.underlying_tokens is None or len(vault_token.underlying_tokens) == 0:
+                return None
+
+            underlying_token = self.base.get_or_create_evm_token(
+                address=vault_token.underlying_tokens[0].address,
+            )
+        except (NotERC20Conformant, NotERC721Conformant) as e:
+            log.error(f'Failed to get tokens for Cur vault {vault_address} due to {e!s}')
```

Comment:
Cur vault? What's that? Please don't use such abbreviations

```suggestion
            log.error(f'Failed to get tokens for Cur vault {vault_address} due to {e!s}')
```